### DISCUSSION TITLE
How to secure all Integration inside a protect using keycloak

### DISCUSSION THREAD

**Discussion by aijaiswa** [apache/camel-karavan#1025] [Q&A]
Hi team, 

GM 

I am trying to secure all integration inside a project using keycloak api. I am able to secure individual integration by calling keycloak API, however there are too much of boiler plated code. 
I want to create a interceptor or filter kind of layer for project, to secure all integrations from a single location. Kindly suggest how I can do that. 
If there are any better approach, please advise

**Comment by mgubaidullin** [comment]
As an option you could use authentication of ingress controller

https://kubernetes.github.io/ingress-nginx/examples/auth/oauth-external-auth/

**Comment by mgubaidullin** [comment]
@aijaiswa could you please share an example of your approach of using keycloak?

**Comment by davsclaus** [comment]
closing er user did not respond
